qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1007,6 +1009,34 @@ def num_tensor_factors(self):
         return rx.number_weakly_connected_components(self._multi_graph)
 
     def __eq__(self, other):
+        """Check if two circuits are equal.
+        This will check .cablirations and .global_phrase and
+        will not consider .data, .name, .metadata when determining.
+        The reason for this is due to being hard and
+        computationally expensive to perform.
+        Please refer to :class:`.Operator` if you need to achieve that level of comparison.
+
+        Examples:
+
+            .. code-block:: python
+
+                from qiskit import QuantumCircuit
+                from qiskit.dagcircuit import DAGCircuit
+
+                qc = qk.QuantumCircuit(2)
+                qc.x(0)
+                qc.x(1)
+                dc = circuit_to_dag(qc);
+
+                qc1 = qk.QuantumCircuit(2)
+                qc1.x(1)
+                qc1.x(0)
+                dc1 = circuit_to_dag(qc1);
+
+                DAGCircuit.__eq__(dc, dc1)
```

Comment:
for code examples that you want to show an output for I recommend using the `.. testcode:` and `.. testoutput` directives. there's some explanation [here](https://qiskit.github.io/qiskit_sphinx_theme/sphinx_guide/how_to_add_code.html#testable-cells) on how to use these directives, and you can check out some examples in our current docs in `docs/migration_guides/algorithms_migration.rst`

Reply:
Thanks for the doc links and examples! got this changed locally.